owncloudApp/src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.kt

```diff
@@ -0,0 +1,467 @@
+        account = requireArguments().getParcelable(ARG_ACCOUNT)
+        if (savedInstanceState != null) {
+            file = savedInstanceState.getParcelable(FileActivity.EXTRA_FILE)
+            account = savedInstanceState.getParcelable(FileActivity.EXTRA_ACCOUNT)
+        }
+
+        return inflater.inflate(R.layout.file_details_fragment, container, false).apply {
+            filterTouchesWhenObscured = PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(context)
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        progressController = TransferProgressController(activity as ComponentsGetter?).also {
+            it.setProgressBar(view.findViewById<ProgressBar>(R.id.fdProgressBar))
+        }
+
+        updateFileDetails(false, false)
+
+        view.findViewById<View>(R.id.fdCancelBtn).setOnClickListener {
+            (mContainerActivity as FileDisplayActivity).cancelTransference(file)
+            fileDetailsViewModel.cancelCurrentDownload(file)
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        outState.putParcelable(FileActivity.EXTRA_FILE, file)
+        outState.putParcelable(FileActivity.EXTRA_ACCOUNT, account)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        progressController?.startListeningProgressFor(file, account)
+        startObservingProgressForDownload()
+    }
+
+    override fun onStop() {
+        progressController?.stopListeningProgressFor(file, account)
+        stopObservingWorkers()
+        super.onStop()
+    }
+
+    private fun startObservingProgressForDownload() {
+        val safeAccount = account ?: return
+
+        fileDetailsViewModel.startListeningToDownloadsFromAccountAndFile(file = file, account = safeAccount)
+        fileDetailsViewModel.pendingDownloads.observe(this) { }
+        fileDetailsViewModel.ongoingDownload.observeWorkerTillItFinishes(
+            owner = this,
+            onWorkEnqueued = { handleDownloadEnqueued() },
+            onWorkRunning = { progress -> handleRunningDownload(progress) },
+            onWorkSucceeded = { handleSuccessDownload() },
+            onWorkFailed = { showMessageInSnackbar(getString(R.string.downloader_download_failed_ticker)) },
+            onWorkCancelled = { showMessageInSnackbar(getString(R.string.downloader_download_canceled_ticker)) },
+            removeObserverAfterNull = false
+        )
+    }
+
+    private fun handleDownloadEnqueued() {
+        requireView().findViewById<View>(R.id.fdProgressBlock).isVisible = true
+
+        requireView().findViewById<TextView>(R.id.fdProgressText).apply {
+            isVisible = true
+            text = String.format(getString(R.string.downloader_download_enqueued_ticker), file.fileName)
+        }
+    }
+
+    private fun handleRunningDownload(progress: Int) {
```

Comment:
With these methods naming seems that the fragment is taking too many responsibilities like handling downloads when you are only updating the progress for example.

`private fun updateDownloadProgress(progress: Int) { }`

Reply:
And same for the other handle methods